packages/vaex-core/vaex/dataframe_protocol.py

```diff
@@ -0,0 +1,489 @@
+        self._col = column
+        
+        # Store the info about category
+        self.is_cat = metadata["vaex.cetagories_bool"][self._col.expression] #is column categorical
+        if metadata["vaex.cetagories"]:
+            self.labels = metadata["vaex.cetagories"][self._col.expression] #list of categories/labels
+        else:
+            self.labels = metadata["vaex.cetagories"]
+    
+    @property
+    def size(self) -> int:
+        """
+        Size of the column, in elements.
+        """
+        return self._col.values.size
+    
+    @property
+    def offset(self) -> int:
+        """
+        Offset of first element. Always zero.
+        """
+        return 0
+        
+    @property
+    def dtype(self) -> Tuple[enum.IntEnum, int, str, str]:
+        """
+        Dtype description as a tuple ``(kind, bit-width, format string, endianness)``
+        Kind :
+            - INT = 0
+            - UINT = 1
+            - FLOAT = 2
+            - BOOL = 20
+            - STRING = 21   # UTF-8
+            - DATETIME = 22
+            - CATEGORICAL = 23
+        Bit-width : the number of bits as an integer
+        Format string : data type description format string in Apache Arrow C
+                        Data Interface format.
+        Endianness : current only native endianness (``=``) is supported
+        Notes:
+            - Kind specifiers are aligned with DLPack where possible (hence the
+              jump to 20, leave enough room for future extension)
+            - Masks must be specified as boolean with either bit width 1 (for bit
+              masks) or 8 (for byte masks).
+            - Dtype width in bits was preferred over bytes
+            - Endianness isn't too useful, but included now in case in the future
+              we need to support non-native endianness
+            - Went with Apache Arrow format strings over NumPy format strings
+              because they're more complete from a dataframe perspective
+            - Format strings are mostly useful for datetime specification, and
+              for categoricals.
+            - For categoricals, the format string describes the type of the
+              categorical in the data buffer. In case of a separate encoding of
+              the categorical (e.g. an integer to string mapping), this can
+              be derived from ``self.describe_categorical``.
+            - Data types not included: complex, Arrow-style null, binary, decimal,
+              and nested (list, struct, map, union) dtypes.
+        """
+        # Define how _dtype_from_vaexdtype calculates kind
+        # If it is internal, categorical must stay categorical (info in metadata)
+        # If it is external (call from_dataframe) must give data dtype
+        
+        bool_c = False
+        if self.is_cat:
+            bool_c = True # internal, categorical must stay categorical
+
+        dtype = self._col.dtype
+            

```

Comment:
```suggestion
            
if dtype.is_encoded:^M
            dtype = dtype.index_type
```
Something like this? If the array is an arrow array, it's the type of the indices you are after right?

Reply:
This works, thank you! Added in new commit.